flixel/input/gamepad/FlxGamepad.hx

```diff
@@ -43,18 +41,67 @@ class FlxGamepad implements IFlxDestroyable
 	 */
 	private var axis:Array<Float> = [for (i in 0...6) 0];
 	
-	#if flash
+	#if (flash || next)
 	private var _device:GameInputDevice; 
 	#end
 	
-	public function new(ID:Int, GlobalDeadZone:Float = 0) 
+	#if (flash)
+	private var _isChrome:Bool = false;
+	#end
+	
+	/**
+	 * Helper class to check if a button is pressed.
+	 */
+	public var pressed:FlxGamepadButtonList;
+	/**
+	 * Helper class to check if a button was just pressed.
+	 */
+	public var justPressed:FlxGamepadButtonList;
+	/**
+	 * Helper class to check if a button was just released.
+	 */
+	public var justReleased:FlxGamepadButtonList;
+	/**
+	 * Helper class to get the float value of analog input.
+	 */
+	public var analog:FlxGamepadAnalogList;
+	
+	public function new(ID:Int, GlobalDeadZone:Float = 0, ?Model:GamepadModel) 
 	{
 		id = ID;
 		
+		model = Model != null ? Model : Xbox;
+		
+		#if flash
+		_isChrome = (Capabilities.manufacturer == "Google Pepper");
+		#end
+		
+		buttonIndex = new ButtonIndex(model);
+		
 		if (GlobalDeadZone != 0)
 		{
 			deadZone = GlobalDeadZone;
 		}
+		
+		pressed = new FlxGamepadButtonList(FlxInputState.PRESSED, this);
+		justPressed = new FlxGamepadButtonList(FlxInputState.JUST_PRESSED, this);
+		justReleased = new FlxGamepadButtonList(FlxInputState.JUST_RELEASED, this);
+		analog = new FlxGamepadAnalogList(this);
+	}
+	
+	public inline function btnID(RawID:Int):ButtonID
```

Comment:
They're modeled exactly after what we have in FlxG.keys.

You do:
```FlxG.keys.justPressed.A```

Likewise here you do:

```gamepad.justPressed.A```